src/SldStyleParser.ts

```diff
@@ -510,121 +519,389 @@ export class SldStyleParser implements StyleParser<string> {
+
+    if (labelEl) {
+      textSymbolizer.label = this.getTextSymbolizerLabelFromSldSymbolizer(labelEl);
     }
-    if (rotation) {
-      markSymbolizer.rotate = parseFloat(rotation);
+
+    textSymbolizer.color = color ? color : '#000000';
+    textSymbolizer.opacity = opacity ? Number(opacity) : 1;
+
+    const haloRadius = get(sldSymbolizer, 'Halo.Radius.#text');
+    if (haloRadius) {
+      textSymbolizer.haloWidth = Number(haloRadius);
     }
-    if (size) {
-      markSymbolizer.radius = parseFloat(size) / 2;
+    if (haloColor) {
+      textSymbolizer.haloColor = haloColor;
     }
-
-    switch (wellKnownName) {
+    const displacement = get(sldSymbolizer, 'LabelPlacement.PointPlacement.Displacement');
+    if (displacement) {
+      const x = get(displacement, 'DisplacementX.#text');
+      const y = get(displacement, 'DisplacementY.#text');
+      textSymbolizer.offset = [
+        x ? parseFloat(x) : 0,
+        y ? parseFloat(y) : 0,
+      ];
+    }
+    const rotation = get(sldSymbolizer, 'LabelPlacement.PointPlacement.Rotation.#text');
+    if (rotation) {
+      textSymbolizer.rotate = Number(rotation);
+    }
+    if (fontFamily) {
+      textSymbolizer.font = [fontFamily];
+    }
+    if (fontStyle) {
+      textSymbolizer.fontStyle = fontStyle as 'normal' | 'italic' | 'oblique' | undefined;
+    }
+    if (fontWeight) {
+      textSymbolizer.fontWeight = fontWeight as 'normal' | 'bold' | undefined;
+    }
+    if (fontSize) {
+      textSymbolizer.size = Number(fontSize);
+    }
+    return textSymbolizer;
+  }
+
+  /**
+   * Create a template string from a TextSymbolizer Label element.
+   * The ordering of the elemments inside the Label element is preserved.
+   *
+   * Examples:
+   * <Label>
+   *  <Literal>foo</Literal>
+   *  <PropertyName>bar</PropertyName>
+   * </Label>
+   * --> "foo{{bar}}"
+   *
+   * <Label>
+   *  <PropertyName>bar</PropertyName>
+   *  <Literal>foo</Literal>
+   * </Label>
+   * --> "{{bar}}foo"
+   *
+   * <Label>
+   *  <PropertyName>bar</PropertyName>
+   *  <Literal>foo</Literal>
+   *  <PropertyName>john</PropertyName>
+   * </Label>
+   * --> "{{bar}}foo{{john}}"
+   *
+   * <Label>
+   *  <PropertyName>bar</PropertyName>
+   *  <PropertyName>john</PropertyName>
+   *  <Literal>foo</Literal>
+   * </Label>
+   * --> "{{bar}}{{john}}foo"
+   *
+   * <Label>
+   *  <PropertyName>bar</PropertyName>
+   *  <PropertyName>john</PropertyName>
+   *  <Literal>foo</Literal>
+   *  <PropertyName>doe</PropertyName>
+   * </Label>
+   * --> "{{bar}}{{john}}foo{{doe}}"
+   *
+   */
+  getTextSymbolizerLabelFromSldSymbolizer = (sldLabel: any): string => {
+    const label: string = sldLabel
+      .map((labelEl: any) => {
+        // TODO: ogc namespace should be removed on parsing. check why it is not
+        // TODO: fast-xml-parser trims everything between xml tags. whitspaces get
+        // lost. When trimValues is set to false ALL whitespaces are kept including newlines etc.
```

Comment:
I'd consider this a bug in our software, even if we have a way around. Can we have an issue, so that we do not forget about this?

Reply:
This TODO is actually solved with the CDATA